libs/estdlib/src/lists.erl

```diff
@@ -523,20 +537,16 @@ unique(Sorted) ->
     unique(Sorted, fun(X, Y) -> X =< Y end).
 
 %% @private
-unique(Sorted, Fun) ->
-    unique(Sorted, Fun, []).
-
-%% @private
-unique([], _Fun, []) ->
+unique([], _Fun) ->
     [];
-unique([X], _Fun, Acc) ->
-    lists:reverse([X | Acc]);
-unique([X, Y | Tail], Fun, Acc) ->
+unique([X], _Fun) ->
+    [X];
+unique([X, Y | Tail], Fun) ->
     case Fun(X, Y) andalso Fun(Y, X) of
         true ->
-            unique([Y | Tail], Fun, Acc);
+            unique([Y | Tail], Fun);
         false ->
-            unique([Y | Tail], Fun, [X | Acc])
+            [X | unique([Y | Tail], Fun)]
```

Comment:
Is this any longer tail recursive?

Reply:
Old version was calling reverse which is not tail recursive.
New version is not tail recursive.